---
title: "Rabies Accessible Genomic Epidemiology (RAGE) Workshop"
author: "Kirstyn Brunker"
date: "2023-05-31"
output:
  html_document:
    toc: true
    toc_float: true
    number_sections: true
    theme: spacelab
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# [Day 1: Introduction and Wet Lab](#1-day1)
## [Genomic surveillance for rabies virus](#1.1)
## [Introduction to nanopore sequencing](#1.2)
## [Masterclass sessions](#1.2)

# [Day 2: Wet lab](#2-day2)
## [Explainer: End prep and barcoding](#2.1)
## [Explainer: Adapter ligation](#2.2)
## [Explainer: DNA quantification](#2.3)
## [Explainer: RAMPART](#2.4)
## [Explainer: MinKNOW](#2.5)
## [RABV seq in practice: Essel/Gurdeep](#2.6)

# [Day 3: Intro to Bioinformatics](#3-day3)
## [Genetic data]()
## [Command line basics]()
## [ARTIC bioinformatics pipeline]()
## [QC, coverage plots and statistics]()

# [Day 4: Bioinformatics: Basic phylogenies & interpretation](#4-day4)
## [ALignments]()
## [Tree explainer]()
## [Building a tree](https://github.com/josephhughes/viral-bioinformatics-training/blob/main/Phylogenetics/4.2.1b_tree_building.md)
## [Tree visualisation]()

# [Day 5: Bioinformatics: Further interpretation and visualisation](#5-day5)
## [MADDOG]()
## [MicroReact]()

